src/tasks/block.rs

```diff
@@ -224,14 +227,14 @@ impl BlockBuilder {
         Ok(())
     }
 
-    /// Simulates a transaction by calling it on the host provider at the current height with the current state.
+    /// Simulates a rollup transaction by calling it on the ru provider at the current height with the current state.
     async fn simulate_transaction(
         &self,
-        host_provider: &Provider,
+        ru_provider: &WalletlessProvider,
         tx_env: TxEnvelope,
     ) -> eyre::Result<()> {
         let tx = TransactionRequest::from_transaction(tx_env);
-        host_provider.call(&tx).await?;
+        ru_provider.call(&tx).await?;
```

Comment:
so this is not entirely correct: bundles need to respect certain parameters they're sent with (which can be inspected by looking at the inner `EthSendBundle`) such as the block number, min/max timestamp, so on and so forth. some txs are also allowed to revert.

I'd say let's hold off from simulating and reconvene along with @prestwich on the sim engine to have an integration plan for this, as we cannot, at the moment, simulate bundles.